cartoframes/data/dataset/registry/dataframe_dataset.py

```diff
@@ -192,42 +148,47 @@ def _is_null(val):
         return vnull.all()
 
 
-def _normalize_column_names(df):
-    column_names = [c for c in df.columns if c not in Column.RESERVED_COLUMN_NAMES]
-    normalized_columns = normalize_names(column_names)
+def _process_columns(df, with_lnglat=None):
+    geom_column = _get_geom_col_name(df)
+    geom_type, enc_type = _get_geometry_type(df, geom_column)
 
-    column_tuples = [(norm, orig) for orig, norm in zip(column_names, normalized_columns)]
+    columns = []
+    for c in df.columns:
+        if c.lower() in Column.FORBIDDEN_COLUMN_NAMES or (with_lnglat and c == geom_column):
+            continue
 
-    changed_cols = '\n'.join([
-        '\033[1m{orig}\033[0m -> \033[1m{new}\033[0m'.format(
-            orig=orig,
-            new=norm)
-        for norm, orig in column_tuples if norm != orig])
+        columns.append({
```

Comment:
Not a big fan of this dictionary.

Given that we have a `Column` class maybe we could use it or subclass it. That way we can encapsulate all that column guessing stuff there. It will help to clean some stuff on this class.

Again, this is 100% nice to have, I'm fine with it as long as it's something private.

Reply:
I thought about creating a new class for that, but I was lazy. I will do it

Reply:
Done